docs/decisions/0017-generalized-containers.rst

```diff
@@ -0,0 +1,95 @@
+17. Modeling Containers as a Generalized Capability for Holding Content
+========================================================================
+
+Context
+-------
+
+This ADR proposes a model for containers that can hold different types of content and can be used to model other content types with similar behavior, such as units, subsections, sections, or courses. The model defines containers' core structure and purpose, the types of containers, content constraints, container children, version control, publishing, and pruning.
+
+Key Concepts
+============
+
+The following definitions provide context for the terms used in this ADR:
+
+- Container: A content structure that holds other content types in a parent-child hierarchy (e.g., sections, subsections, units).
+- PublishableEntity: A versioned content type that supports publishing and can exist independently or within a container (e.g., sections, units, components).
+- Versioning: The process of creating new container versions when its structure or metadata changes (e.g., title, order, adding/removing children).
+- Deletion: Removing a draft or published entity from the learning package. This is done by setting its version to NULL, while keeping historical versions for potential reverts.
+- Removal from a container: Removing an entity from a parent container, without removing it from the learning package.
+- Publishing: Making a container and its content available in a published state.
+- Pruning: Removing unused, unpublished container versions if they are not the latest and have no references.
+- Selectors: A mechanism for dynamically selecting multiple PublishableEntities based on defined criteria (e.g., split testing, randomized content selection).
+
+Decisions
+---------
+
+1. Core Structure and Purpose of Containers
+===========================================
+
+This section defines the purpose and structure of containers, explaining how they are designed to hold various types of content through a parent-child setup.
+
+- A container is designed as a generalized capability to hold different types of content.
+- A container is a publishable content type that holds other content types through a parent-child relationship. For example, sections, subsections and units.
+- The generalized container capability will have its own Django application as part of the authoring application allowing other types of containers and content types will build on top of. For instance:
```

Comment:
Containers were folded into the `publishing` app because of the current coupling that exists between drafts and publishing between children and parents. The alternative would have been to do everything via signals, but we decided that it wasn't worth the added complexity and potential for confusion.

Reply:
Updated! Thank you